src/lib/core/match-media/match-media.ts

```diff
@@ -43,6 +43,15 @@ export class MatchMedia {
     return results;
   }
 
+  get registry(): Map<string, MediaQueryList> {
+    return this._registry;
+  }
+
+  set registry(registry: Map<string, MediaQueryList>) {
+    this.emptyRegistry();
```

Comment:
I think this is problematic. Consider the scenario where `MediaTrigger` "steals" ownership of the registry to force different activations. This would empty the registry here and unregister the listener functions, however when the `MediaTrigger` then restores the registry using the media queries it set aside, those will effectively be "dead" as no more listeners are registered.

Either `MediaTrigger` needs to take over full ownership and be made responsible for its own cleanup, or cleanup needs to be managed separately from activations. I think the latter is the simplest approach, with the catch that "borrowed" media queries in `MediaTrigger` will be unregistered when destroying `MatchMedia`. I think this is fine though, as `MediaTrigger` itself doesn't require the `MediaQueryList` to have a listener, *and* the listeners were registered in `MatchMedia` in the first place.

I prototyped this in https://github.com/JoostK/flex-layout/commit/38e26679387038e76901214949878d764ea4e1fd, which is a fixup commit of this PR's commit. With that approach, the getter/setter dance of `this.registry` is no longer strictly required, nor are the changes to `MediaTrigger.restoreRegistryMatches`.